qt/scientific_interfaces/Indirect/InelasticDataManipulationIqtTabView.cpp

```diff
@@ -0,0 +1,437 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2018 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include "InelasticDataManipulationIqtTabView.h"
+#include "MantidAPI/AlgorithmManager.h"
+#include "MantidAPI/ITableWorkspace.h"
+#include "MantidGeometry/Instrument.h"
+#include "MantidQtWidgets/Common/AlgorithmRuntimeProps.h"
+#include "MantidQtWidgets/Common/QtPropertyBrowser/qteditorfactory.h"
+#include "MantidQtWidgets/Common/SignalBlocker.h"
+#include "MantidQtWidgets/Common/UserInputValidator.h"
+#include "MantidQtWidgets/Plotting/RangeSelector.h"
+
+#include <QFileInfo>
+
+#include <algorithm>
+
+#include "IndirectAddWorkspaceDialog.h"
+
+using namespace Mantid::API;
+using namespace MantidQt::API;
+using namespace MantidQt::MantidWidgets;
+
+namespace {
+
+/**
+ * Calculate the number of bins in the sample & resolution workspaces
+ * @param wsName The sample workspace name
+ * @param resName the resolution woskapce name
+ * @param energyMin Minimum energy for chosen bin range
+ * @param energyMax Maximum energy for chosen bin range
+ * @param binReductionFactor The factor by which to reduce the number of bins
+ * @return A 4-tuple where the first entry denotes whether the
+ * calculation was successful or not. The final 3 values
+ * are EWidth, SampleBins, ResolutionBins if the calculation succeeded,
+ * otherwise they are undefined.
+ */
+std::tuple<bool, float, int, int> calculateBinParameters(std::string const &wsName, std::string const &resName,
+                                                         double energyMin, double energyMax,
+                                                         double binReductionFactor) {
```

Comment:
In order to be consistent with the MVP pattern, I think this should be in the model because it is logic (running algorithm) and not UI related. The presenter can call `updateDisplayedBinParameters` with parameters returned from this function. I guess this is similar to the way preview is implemented in symmetrize tab

Reply:
I would agree except for in the case where view is calling this directly when values are changed.

Reply:
Yeah, right it doesn't violate the pattern because the UI is not coupled to this function. I suggest it is better to put it in the presenter instead of being a standalone function.